Source/WebCore/page/LocalFrameView.cpp

```diff
@@ -4677,8 +4677,9 @@ void LocalFrameView::setWasScrolledByUser(bool wasScrolledByUser)
     if (currentScrollType() == ScrollType::Programmatic)
         return;
 
-    if (wasScrolledByUser)
-        m_frame->document()->setGotoAnchorNeededAfterStylesheetsLoad(false);
+    auto document = m_frame->document();
```

Comment:
```suggestion
    auto document = m_frame->protectedDocument();
```

Reply:
Or alternately:
```suggestion
    RefPtr document = m_frame->document();
```
(not sure which one we prefer…)

Reply:
We prefer:
```
RefPtr document = m_frame->document();
```

The `protectedFoo()` are used when there is no local variable.